### PR TITLE
feat(table): toolbar is not shown when only row actions are present

### LINKED ISSUE #1423: [Table] Row actions cause empty toolbar
### Describe the bug

If RowActions are being used in a Table, the Table renders an empty toolbar, even if there is no toolbar content defined.

### To Reproduce

Go to https://carbon-addons-iot-react.com/?path=/story/watson-iot-table--with-row-expansion-and-actions

Notice the blank toolbar, due to the Table having row actions.

### Expected behavior

There should be no toolbar if there is no toolbar content defined.

Screen Shot 2020-07-22 at 2 07 45 PM

### PR DESCRIPTION
**Pull request by marcJV**
Closes #1423

**Summary**

Row actions no longer render a toolbar, also added if multiselect is present and there is a selection, the toolbar is shown.

**Change List (commits, features, bugs, etc)**

- Removes the check for row actions when rendering the toolbar, but adds in check for multiselect and selection

**Acceptance Test (how to verify the PR)**

- Verify that Table 'with row expansion and actions' no longer supplies an empty toolbar
- Verify that there are no other row actions that require toolbar support

### CODE REVIEW

**Comment by stuckless** [comment]
Is this a regressions 
image

I'm noticing stories with row details are providing a misalignment of the table header with selections

**Comment by marcJV** [comment]
> Is this a regressions
> image
> 
> I'm noticing stories with row details are providing a misalignment of the table header with selections

This is also showing up in master from another PR, this may require some triage though is already part of: #1597. The misalignment may need to be a separate issue.

File: src/components/Table/Table.story.jsx [hunk 1]
```diff
@@ -2247,7 +2247,7 @@ storiesOf('Watson IoT/Table', module)
   .add(
     'with row expansion and actions',
     () => (
-      <Table
+      <StatefulTable
```
Comment (tay1orjones) [inline src/components/Table/Table.story.jsx:2250]: Why switch this one from Table to StatefulTable?

**Review by tay1orjones — APPROVED** [review]
Looks good pending my one comment - I think we can leave the story as is

**Review by davidicus — APPROVED** [review]

**Pull request merged**